### PR TITLE
Add pipeline to generate valid SDK change against WIP codegen change

### PR DESCRIPTION
**Pull request by pshao25**
Steps:
1. Publish three packages (codegen nuget package, @autorest/csharp, @azure-tools/cadl-csharp) to internal feed with alpha version to differentiate it from official build. And put these packages into the build artifacts. Nuget packages are published to https://dev.azure.com/azure-sdk/public/_artifacts/feed/azure-sdk-for-net, npm packages are published to https://dev.azure.com/azure-sdk/public/_artifacts/feed/azure-sdk-for-js-test-autorest currently, we will change to https://dev.azure.com/azure-sdk/public/_artifacts/feed/azure-sdk-for-js later.
2. Regen sdk code in parallel.

Sample pipeline: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2641887&view=results
Sample PR: https://github.com/Azure/azure-sdk-for-net/pull/35005

### CODE REVIEW

**Review by m-nash — COMMENTED** [review]
Lets maybe separate this into a couple scripts that can be stitched together.

`Update-Local-References.ps1` --responsible for editing package.data.props, etc
`Regen-Using-Local-Generator.ps` --this first calls `Update-Local-References.ps1` then calls generate / test with possible a flag to skip testing if you just want the generate first.

This separation will make it easier to update the `Regen` portion to have a "start where we left off feature". 

If you run the command and it fails on the 158th library out of 180 and you make a quick change sometimes you just want to start where you left off before doing another full run.

File: eng/LocalUpdateAzureSdkForNet.ps1 [hunk 1]
```diff
@@ -0,0 +1,60 @@
+param($SdkRepoRoot)
+
+$SdkRepoRoot = Resolve-Path $SdkRepoRoot
+$repoRoot = Resolve-Path (Join-Path $PSScriptRoot '..')
+$emitterPath = Resolve-Path (Join-Path $repoRoot 'src' 'CADL.Extension' 'Emitter.Csharp')
+
+# Pack local autorest.csharp
+$currentDate = $(Get-Date -UFormat "%Y%m%d")
+Push-Location $repoRoot
+dotnet pack /p:OfficialBuildId=$currentDate.100
```
Comment (chunyu3) [inline eng/LocalUpdateAzureSdkForNet.ps1:10]: Don't hard code `100`. Maybe an parameter to set it.

File: eng/LocalUpdateAzureSdkForNet.ps1 [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+Pop-Location
+$packagePath = Resolve-Path (Join-Path $repoRoot "artifacts" "packages" "Debug" "Microsoft.Azure.AutoRest.CSharp.3.0.0-beta.$currentDate.100.nupkg")
```
Comment (chunyu3) [inline eng/LocalUpdateAzureSdkForNet.ps1:12]: Non-used variable?

**Review by archerzz — COMMENTED** [review]
Part of the logic has already been implemented in https://github.com/Azure/autorest.csharp/blob/feature/v3/eng/UpdateAzureSdkForNet.ps1 which will update the generator and emitter versions, then generate the codes. I suggest to update the script to only build and package (which might already been implemented elsewhere), then invoke `UpdateAzureSdkForNet.ps1`.

**Comment by archerzz** [comment]
A minor suggestion I just thought of: we could add `[Preview]` to the PR title, and create the PR as draft with `do not merge` tag.

**Comment by m-nash** [comment]
@hallipr @mikeharder FYI

**Comment by pshao25** [comment]
beta -> dev
nuget -> xxx.alpha.xxx -> azure-sdk-for-net
npm(xxx.beta.xxx) -> xxx.(ask js) ->  azure-sdk-for-net (ask js)

0.1.11

internal project:
name: Autorest Regen Preview
PR title:  Autorest Regen Preview to xxx.alpha.xxx //Make it draft/ [DoNotMerge] /description to pull request

**Review by m-nash — APPROVED** [review]

**Pull request merged**